File: euler-ode/bin/dpm/filter-orbit.py
# run data filter as `python filter-orbit.py`
import sys

print("t", "\t", "q1", "\t", "p1", "\t", "dE")
for line in sys.stdin:
    a = line.split(" ")
    t = a[0]
    q = a[1].split(",")
    p = a[2].split(",")
    if not 'e0' in vars():
        e0 = float(a[3])
    print(t, "\t", q[0], "\t", p[0], "\t", float(a[3]) - e0)

File: euler-ode/bin/cr3bp/filter-orbit.js
/* run data filter as `node filter-orbit.js` */
var readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    terminal: false
});

var e0;
console.log("%s\t%s\t%s\t%s", 't', 'x', 'y', 'dE');
rl.on('line', function (line) {
    var a = line.split(" ");
    var t = a[0], coord = a[1], e = a[3];
    var q = coord.split(",");
    if(!e0) { e0 = e; }
    console.log("%s\t%s\t%s\t%s", t, q[0], q[1], e - e0);
});
